### ISSUE TITLE
SC1072/1073 error with >/dev/null in if statement, v0.4.6

### ISSUE THREAD

**Issue by aumars** [koalaman/shellcheck#872]
#### For bugs
- [x] I tried on shellcheck.net and verified that this is still a problem on the latest commit
- [ ] It's not reproducible on shellcheck.net, but I think that's because it's an OS, configuration or encoding issue

#### Here's a snippet or screenshot that shows the problem:

```sh
#!/bin/bash
command="pacman -Qs sdl2"

if [ -n "$command" >/dev/null ]
...
```
#### Here's what shellcheck currently says:
```
$ shellcheck myscript
 
Line 5:
if [ -n "$command" >/dev/null ]
^-- SC1009: The mentioned parser error was in this if expression.
   ^-- SC1073: Couldn't parse this test expression.
                   ^-- SC1072: Expected test to end here (don't wrap commands in []/[[]]). Fix any mentioned problems and try again.

$ 
```

#### Here's what I wanted or expected to see:

I'm using `>/dev/null` in my `if` statement and it's apparently viewed as an error. However, the program works just as desired. Any solution?

**Comment by brother** [comment]
> However, the program works just as desired.

Well. Is it really?

You are validating that the length of the command string is nonzero, it clearly is and the test i moot. What are you actually trying to do?

```
#!/bin/bash
if pacman -Qs sdl2 >/dev/null; then
echo "hooray"
fi
```

Or what?

**Comment by aumars** [comment]
For some reason, [linter-shellcheck](https://atom.io/packages/linter-shellcheck) doesn't pick up the error anymore on my Atom editor. I guess this is a false flag. Sorry @brother if I had wasted your time.

Anyways, here is the full script I'm currently using.
```
#!/bin/bash

if [ $1 = "1" ]
  then
    command=""
    install=""
    compile=""
elif [ $1 = "2" ]
  then
    command='pacman -Qs sdl2'
    install='sudo pacman -S sdl2'
    compile='make all'
else
  echo Warning! No arguments detected! Shell script is possibly corrupt!
  exit;
fi

if [ -n "$command" >/dev/null ]
  then
    echo sdl2 package installed!
    $compile
    if [ -e caventure ]
      then
        echo Installation complete!
        echo Would you like to run it now? [Y/N]
        select yn in "Yes" "No"; do
          case $yn in
            Yes ) ./caventure; break;;
            No ) exit;;
          esac
        done
    else
      echo Warning! Build failed!
    fi
else
    echo sdl2 package not installed!
    echo Would you like to install it now? [Y/N]
    select yn in "Yes" "No"; do
      case $yn in
        Yes ) $install; break;;
        No ) exit;;
      esac
    done
fi
```

**Comment by brother** [comment]
np.
Still a bit puzzled over the test.

```shell
brother ~$ command="false"
brother ~$ if [ -n "$command" > /dev/null ]; then echo "hello"; else echo "nope"; fi
hello
brother ~$ command="date"
brother ~$ if [ -n "$command" > /dev/null ]; then echo "hello"; else echo "nope"; fi
hello
```

My suggestion from earlier:

```shell
brother ~$ command="date"
brother ~$ if "$command" > /dev/null; then echo "hello"; else echo "nope"; fi
hello
brother ~$ command="false"
brother ~$ if "$command" > /dev/null; then echo "hello"; else echo "nope"; fi
nope
```
